jujugui/static/gui/src/app/app.js

```diff
@@ -1217,6 +1217,14 @@ YUI.add('juju-gui', function(Y) {
         document.getElementById('header-help'));
     },
 
+    _renderHeaderLogo: function() {
+      ReactDOM.render(
+        <window.juju.components.HeaderLogo
+        changeState={this.state.changeState.bind(this.state)}
+        user={this._getAuth()} />,
```

Comment:
This component seems to only require, for instance, a navigateUserProfile callable, defined as:
```
this.state.changeState.bind(this.state, {
  model: null,
  profile: this._getAuth().rootUserName,
  root: null,
  store: null
});
```
This way the component can be agnostic about the current implementation of the state.